controllers/vpcendpoint/helpers.go

```diff
@@ -561,3 +560,39 @@ func tagsContains(tags []ec2Types.Tag, tagsToCheck map[string]string) bool {
 
 	return true
 }
+
+// createMissingPrivateZoneTags will compare existing tags to the required set and apply if missing
+func (r *VpcEndpointReconciler) createMissingPrivateZoneTags(ctx context.Context, zoneID string) error {
+	// Find existing tags
+	listTagsOut, err := r.awsClient.FetchPrivateZoneTags(ctx, zoneID)
+	if err != nil {
+		return fmt.Errorf("failed to list zone's tags %w", err)
+	}
+	// Generate default tags to compare against
+	generatedDefaultTagInput, err := r.awsClient.GenerateDefaultTagsForHostedZoneInput(zoneID, r.clusterInfo.clusterTag)
+	if err != nil {
+		return fmt.Errorf("failed to generate hosted zone's default tags %w", err)
+	}
+
+	defaultTagsMap := map[string]string{}
+	for _, tag := range generatedDefaultTagInput.AddTags {
+		defaultTagsMap[*tag.Key] = *tag.Value
+	}
+
+	for k, v := range defaultTagsMap {
+		found := false
+		for _, tag := range listTagsOut.ResourceTagSet.Tags {
+			if *tag.Key == k && *tag.Value == v {
+				found = true
+				break
+			}
+		}
+		if !found { //Apply Missing Tags
+			_, err := r.awsClient.ChangeTagsForResource(ctx, generatedDefaultTagInput)
+			if err != nil {
+				return fmt.Errorf("failed tag hosted zone with default tags %w", err)
+			}
+		}
+	}
```

Comment:
It's a little more straightforward to assemble a map of the actual tags to compare a slice of default tags against, something like (if we make an `actualTagsMap`):

```go
for _, tag := range defaultTags {
	v, ok := actualTagsMap[tag.key]
  // If doesn't exist or the value doesn't match the expected default tag
	if !ok || v != tag.Value {
	  if _, err := r.awsClient.ChangeTagsForResource(ctx, generatedDefaultTagInput); err != nil {
		return fmt.Errorf("failed tag hosted zone with default tags %w", err)
	}
}
```

Reply:
Thank you, this has been implemented.